docs/weaviate/concepts/data-import.mdx

```diff
@@ -0,0 +1,71 @@
+
+:::tip
+
+For **code examples**, check out the [How-to: Batch import](../manage-objects/import.mdx) guide. Currently, only the Python client supports batch imports.
+
+:::
+
+---
+
+## Server-side batching
+
+:::caution Technical preview
+
+Server-side batching was added in **`v1.33`** as a **technical preview**.<br/><br/>
+This means that the feature is still under development and may change in future releases, including potential breaking changes.
+**We do not recommend using this feature in production environments at this time.**
+
+:::
+
+Weaviate's server-side batching, also known as **automatic batching**, is an intelligent mechanism designed to make data ingestion simpler, faster, and more robust. Instead of manually tuning batch parameters on the client side, you can let the server dynamically manage the data flow for optimal performance.
+
+This mode is a drop-in replacement for `fixed_size` or `dynamic` batching and is the recommended method for importing data.
+
+### How it works
+
+When you use the `automatic()` batching context manager, you initiate a persistent connection to the server for the duration of the batch job.
+
+1.  **Client sends data**: Your client sends objects to the server in chunks. These requests execute instantly without waiting for the data to be fully imported.
+2.  **Server manages queues**: The server places incoming objects into an internal queue system. This decouples the network communication from the actual database work (like vectorization and storage).
+3.  **Dynamic backpressure**: The server continuously monitors its internal queue size. It calculates an exponential moving average (EMA) of its workload and tells the client the ideal number of objects to send in the next chunk. This feedback loop allows the system to self-regulate, maximizing throughput without overwhelming the server.
+4.  **Asynchronous errors**: If an error occurs while processing an object (e.g., validation fails), the server sends the error message back to the client over a separate, dedicated stream without interrupting the flow of other objects.
+
+This architecture centralizes the complex batching logic on the server, resulting in a more efficient and stable data ingestion pipeline for all connected clients.
+
+:::info Why use automatic (server-side) batching?
+
+- **Simplified client code**: No need to tweak `batch_size` and `concurrent_requests` manually. The server determines the optimal batch size based on its current workload.
+- **Improved stability**: The system automatically applies **backpressure**. If the server is busy, it will instruct the client to send less data, preventing overloads and request timeouts, which is especially useful during long-running vectorization tasks.
+- **Enhanced resilience**: It's designed to handle cluster events like node scaling more gracefully, reducing the risk of interrupted batches.
```

Comment:
It's not clear from the docs how this happens. Is it that when a node is down, the server tells the client to slow down?